server/__test__/PostReview.test.js

```diff
@@ -0,0 +1,49 @@
+const reviewService = require('../src/service/ReviewsService');
+const reviewsDao = require('../src/repository/ReviewsDAO');
+
+jest.mock('../src/repository/ReviewsDAO');
+
+//test object
+const testSuccess = {
+    username: "John",
+    imageUrl: "img",
+    rating: 5,
+    content: "Test Success"
+}
+
+//test fail object
+const testFail = {
+    imageUrl: "img",
+    rating: 5,
+    content: "Test Fail"
+}
+
+describe('Test to see if Review is Posted', () => {
+    //test to see success of the 
+    test('Successfully Posted Review', async() => {
+        reviewsDao.postReview.mockResolvedValue(true);
+
+        const result = await reviewService.createNewReview(testSuccess);
+
+        expect(result.author).toBe(testSuccess.username);
+        expect(result.imageUrl).toBe(testSuccess.imageUrl);
+        expect(result.rating).toBe(testSuccess.rating);
+        expect(result.content).toBe(testSuccess.content);
+    })
+
+    //test that fails in the DAO
+    test('Fails when trying to Post in DAO', async() => {
+        reviewsDao.postReview.mockResolvedValue(false);
+
+        const result = await reviewService.createNewReview(testSuccess);
+
+        expect(result).toBeFalsy();
+    })
+
+    //test fails the validate data
+    test('Fails validate data', async() => {
```

Comment:
The tests for `validateFields` function, need to be isolated from anything related to `createNewReview`.  This test should just call `validateFields`.